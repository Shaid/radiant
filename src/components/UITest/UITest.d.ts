export interface IState {
}

export interface IProps {
}

export interface IDefaultProps {
}

export type PropsWithDefaults = IProps & IDefaultProps
