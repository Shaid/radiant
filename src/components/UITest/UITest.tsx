import React from 'react'
import styled from 'theme'

import { IProps, IDefaultProps, IState } from './UITest.d'

const UIContainer = styled.div`
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
`
const RoomDescription = styled.div`
  margin: 2rem;
`

const ActionPanel = styled.div`
  margin: 2rem;
`

const StatusBar = styled.div`
  margin: auto 2rem 2rem 2rem;
`

const AreaName = styled.blockquote`
  font-size: 1.25rem;
  margin: 1rem 0;
`

const RoomName = styled.blockquote`
  font-size: 1.5rem;
  margin: 1rem 0;
`


class UITest extends React.PureComponent<IProps, IState> {
  static defaultProps: Partial<IDefaultProps> = {}

  render() {
    // const {} = this.props as PropsWithDefaults
    return (
      <UIContainer>
        <RoomDescription>
          <AreaName>Villespont, <em>the City of Bridges</em></AreaName>
          <RoomName><strong>Lower Arch Station</strong>, the demilitarised zone</RoomName>
          <p>On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.</p>
          <p>You feel the weight of the bridges above you.</p>
        </RoomDescription>
        <ActionPanel>
          A young lady is here. She seems bored.
        </ActionPanel>
        <StatusBar>
          <p>It is late in the day.</p>
          <p>You feel oppressed, and hungry.</p>
        </StatusBar>
      </UIContainer>
    )
  }
}

export default UITest
