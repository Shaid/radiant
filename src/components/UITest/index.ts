export { default } from './UITest'

// NOTE: uncomment this line to pass through named exports
// export * from './UITest'
