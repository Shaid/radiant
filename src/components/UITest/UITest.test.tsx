import React from 'react'
import { shallow } from 'enzyme'

import UITest from '.'

describe('<UITest />', () => {
  it('renders', () => {
    const wrapper = shallow(<UITest />)
    expect(wrapper).toMatchSnapshot()
  })

  it('TODO: tests for UITest')
})
