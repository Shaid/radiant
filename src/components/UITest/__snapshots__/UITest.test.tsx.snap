// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UITest /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UITest />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <UITest__RoomDescription>
          <UITest__AreaName>
            Villespont, 
            <em>
              the City of Bridges
            </em>
          </UITest__AreaName>
          <UITest__RoomName>
            <strong>
              Lower Arch Station
            </strong>
            , the demilitarised zone
          </UITest__RoomName>
          <p>
            On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
          </p>
          <p>
            You feel the weight of the bridges above you.
          </p>
        </UITest__RoomDescription>,
        <UITest__ActionPanel>
          A young lady is here. She seems bored.
        </UITest__ActionPanel>,
        <UITest__StatusBar>
          <p>
            It is late in the day.
          </p>
          <p>
            You feel oppressed, and hungry.
          </p>
        </UITest__StatusBar>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <UITest__AreaName>
              Villespont, 
              <em>
                the City of Bridges
              </em>
            </UITest__AreaName>,
            <UITest__RoomName>
              <strong>
                Lower Arch Station
              </strong>
              , the demilitarised zone
            </UITest__RoomName>,
            <p>
              On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
            </p>,
            <p>
              You feel the weight of the bridges above you.
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Villespont, ",
                <em>
                  the City of Bridges
                </em>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Villespont, ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "the City of Bridges",
                },
                "ref": null,
                "rendered": "the City of Bridges",
                "type": "em",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <strong>
                  Lower Arch Station
                </strong>,
                ", the demilitarised zone",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lower Arch Station",
                },
                "ref": null,
                "rendered": "Lower Arch Station",
                "type": "strong",
              },
              ", the demilitarised zone",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
            },
            "ref": null,
            "rendered": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You feel the weight of the bridges above you.",
            },
            "ref": null,
            "rendered": "You feel the weight of the bridges above you.",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "A young lady is here. She seems bored.",
        },
        "ref": null,
        "rendered": "A young lady is here. She seems bored.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <p>
              It is late in the day.
            </p>,
            <p>
              You feel oppressed, and hungry.
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "It is late in the day.",
            },
            "ref": null,
            "rendered": "It is late in the day.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You feel oppressed, and hungry.",
            },
            "ref": null,
            "rendered": "You feel oppressed, and hungry.",
            "type": "p",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <UITest__RoomDescription>
            <UITest__AreaName>
              Villespont, 
              <em>
                the City of Bridges
              </em>
            </UITest__AreaName>
            <UITest__RoomName>
              <strong>
                Lower Arch Station
              </strong>
              , the demilitarised zone
            </UITest__RoomName>
            <p>
              On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
            </p>
            <p>
              You feel the weight of the bridges above you.
            </p>
          </UITest__RoomDescription>,
          <UITest__ActionPanel>
            A young lady is here. She seems bored.
          </UITest__ActionPanel>,
          <UITest__StatusBar>
            <p>
              It is late in the day.
            </p>
            <p>
              You feel oppressed, and hungry.
            </p>
          </UITest__StatusBar>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <UITest__AreaName>
                Villespont, 
                <em>
                  the City of Bridges
                </em>
              </UITest__AreaName>,
              <UITest__RoomName>
                <strong>
                  Lower Arch Station
                </strong>
                , the demilitarised zone
              </UITest__RoomName>,
              <p>
                On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.
              </p>,
              <p>
                You feel the weight of the bridges above you.
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Villespont, ",
                  <em>
                    the City of Bridges
                  </em>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Villespont, ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "the City of Bridges",
                  },
                  "ref": null,
                  "rendered": "the City of Bridges",
                  "type": "em",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <strong>
                    Lower Arch Station
                  </strong>,
                  ", the demilitarised zone",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lower Arch Station",
                  },
                  "ref": null,
                  "rendered": "Lower Arch Station",
                  "type": "strong",
                },
                ", the demilitarised zone",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
              },
              "ref": null,
              "rendered": "On a nearby corner a violin sings; the cadence of its bow dancing in time with the rhythm of your thoughts.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You feel the weight of the bridges above you.",
              },
              "ref": null,
              "rendered": "You feel the weight of the bridges above you.",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "A young lady is here. She seems bored.",
          },
          "ref": null,
          "rendered": "A young lady is here. She seems bored.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p>
                It is late in the day.
              </p>,
              <p>
                You feel oppressed, and hungry.
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It is late in the day.",
              },
              "ref": null,
              "rendered": "It is late in the day.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You feel oppressed, and hungry.",
              },
              "ref": null,
              "rendered": "You feel oppressed, and hungry.",
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
